examples/cpp/BasicOperations/main.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// Include the Geode library.
+#include <iostream>
+#include <geode/CacheFactory.hpp>
+#include <geode/PoolManager.hpp>
+// Use the "geode" namespace.
+using namespace apache::geode::client;
+
+
+// The BasicOperations example.
+int main(int argc, char** argv) {
+
+  auto cacheFactory = CacheFactory();
+  cacheFactory.set("log-level", "none");
+  auto cache = cacheFactory.create();
+
+  auto poolFactory = cache.getPoolManager().createFactory();
+  poolFactory->addLocator("localhost", 10334);
+  auto pool = poolFactory->create("pool");
+
+  auto regionFactory = cache.createRegionFactory(CACHING_PROXY);
+
+  auto region = regionFactory.setPoolName("pool").create("exampleRegion");
```

Comment:
These steps are a lot of setup and may be magic APIs for someone new to the codebase.  How about splitting them up into self-documenting functions, createCache and createRegion?

Reply:
Why are we doing a cacheing proxy? That is a pretty advanced use case for a basic example.